vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -493,6 +505,517 @@ protected void fireUpdateProgress(long totalBytes, long contentLength) {
+        }
+
+        public JsonObject getDetail() {
+            return detail;
+        }
+
+        public JsonObject getDetailXhr() {
+            return detailXhr;
+        }
+
+        public JsonObject getDetailFile() {
+            return detailFile;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code upload-start} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    protected Registration addUploadStartListener(
+            ComponentEventListener<UploadStartEvent> listener) {
+        return addListener(UploadStartEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    @DomEvent("upload-success")
+    public static class UploadSuccessEvent extends ComponentEvent<Upload> {
+        private final JsonObject detail;
+        private final JsonObject detailXhr;
+        private final JsonObject detailFile;
+
+        public UploadSuccessEvent(Upload source, boolean fromClient,
+                @EventData("event.detail") JsonObject detail,
+                @EventData("event.detail.xhr") JsonObject detailXhr,
+                @EventData("event.detail.file") JsonObject detailFile) {
+            super(source, fromClient);
+            this.detail = detail;
+            this.detailXhr = detailXhr;
+            this.detailFile = detailFile;
+        }
+
+        public JsonObject getDetail() {
+            return detail;
+        }
+
+        public JsonObject getDetailXhr() {
+            return detailXhr;
+        }
+
+        public JsonObject getDetailFile() {
+            return detailFile;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code upload-success} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    protected Registration addUploadSuccessListener(
+            ComponentEventListener<UploadSuccessEvent> listener) {
+        return addListener(UploadSuccessEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    public static class FilesChangeEvent extends ComponentEvent<Upload> {
+        private final JsonArray files;
+
+        public FilesChangeEvent(Upload source, boolean fromClient) {
+            super(source, fromClient);
+            this.files = source.getFilesJsonArray();
+        }
+
+        public JsonArray getFiles() {
+            return files;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code files-changed} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    protected Registration addFilesChangeListener(
```

Comment:
Didn't find any use for this method, could be removed

Reply:
Removed.